src/lamp.py

```diff
@@ -11,18 +11,38 @@
 class Lamp:
     def __init__(self, name, base_color, shade_color):
         self.name = name
+        self.behaviours = []
         self.shade = LedStrip(shade_color, pixel_config['shade']['pin'], pixel_config['shade']['pixels'])
         self.base = LedStrip(base_color, pixel_config['base']['pin'], pixel_config['base']['pixels'])
+    
+    def add_behaviour(self,behaviour_class):
+        b = behaviour_class(self)
+        print(b)
+        asyncio.create_task(b.run())
+        
+        self.behaviours.append(b) 
```

Comment:
If we want to be able to remove a behaviour, we need to hold onto the `Task` object that `create_task` returns in order to call `cancel()` on it.